src/Build/BuildCheck/Infrastructure/CheckScopeClassifier.cs

```diff
@@ -7,8 +7,13 @@
 using Microsoft.Build.Shared;
 
 namespace Microsoft.Build.BuildCheck.Infrastructure;
-internal static class CheckScopeClassifier
+
+internal class CheckScopeClassifier : IDisposable
```

Comment:
Let's not mix static and instance usages.
The dispose on instance clears subscriptions on static class - that can lead to surprises.

I'd suggest to keep this static and having the subscribers being responsible for deregistering (it's a common practice with events anyways)